### PR TITLE
Extending the excludes/warns format + fixes

### PR DESCRIPTION
**Pull request by waterfoul**
This PR extends the excludes/warns format to also support package@version syntax. Additionally I fixed a bug which blocked the tool from running if you have an invalid peer dependency (there are valid use cases where this is OK). I also added some logic to send the error message from ossindex to the console if a 500 error happens when requesting data

### CODE REVIEW

File: lib/owdit.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ function check(cwd, callback) {
   });
 
   if (result.status !== 0) {  //eslint-disable-line no-magic-numbers
-    return callback(new Error(`Error running npm ls: ${result.stderr}`));
+	console.error(`Error running npm ls: ${result.stderr}`);
```
Comment (KurtPattyn) [inline lib/owdit.js:99]: Why did you remove the callback here?

File: lib/owdit.js [hunk 2]
```diff
@@ -152,40 +152,44 @@ function check(cwd, callback) {
...
-          vulnerabilities: element.vulnerabilities.map((element) => {
-            return {
-              title:       element.title,
-              description: element.description,
-              versions:    element.versions,
-              references:  element.references
-            };
-          })
+    const request = client.post("https://ossindex.net/v2.0/package", restOptions, (data, response) => {
+      if(response.statusCode >= 500) {
```
Comment (KurtPattyn) [inline lib/owdit.js]: It is not clear why this is needed? Could you please add a comment in the code that explains why this is needed (and when it could happen for instance).

**Review by KurtPattyn — COMMENTED** [review]
Thanks for contributing!
I added some questions inside the commits; could you please review?

File: lib/owdit.js (hunk 1 above)
Comment (waterfoul) [reply lib/owdit.js:99]: If you run the callback here an incorrect peer dependency will cause the whole process to bomb out

File: lib/owdit.js (hunk 2 above)
Comment (waterfoul) [reply lib/owdit.js]: Done

File: lib/owdit.js (hunk 1 above)
Comment (waterfoul) [reply lib/owdit.js:99]: To elaborate further, I have a project which depends on `angular-bootstrap-daterange-picker^1.0.2` which in turn has a peer dependency on `jquery@2`. The underlying bootstrap library behind `angular-bootstrap-daterange-picker` updated to support `jquery@3` but the peer has not been updated yet. Everything does work in this case with `jquery@3` so I am ignoring the peer dep warning when I install and it was causing owdit to exit here

**Review by KurtPattyn — APPROVED** [review]

**Comment by waterfoul** [comment]
Any chance this will get merged?

**Comment by KurtPattyn** [comment]
@waterfoul Merged. I still need to tag the branch so a new version is pushed to NPM.

**Comment by KurtPattyn** [comment]
@waterfoul New version has been published (it's now version 1.2.0)
